### PR TITLE
Update licenses for third_party

### LINKED ISSUE #43: Generated files are derivative works
Using a code generator on a file means that the generated file is a derivative work.

We need to do the following:

- [ ] Move the committed generated files to the third_party folder (preferably next to their sources). (Yes, that means our imports are going to be a bit weird.)
- [x] Generate the same copyright string in the top of the file as was in the original file.
  - [x] Either have an option in yaml to put the copyright string,
  - [ ] Or, find a clever way to extract the first comment in a file if it's the copyright string.

@jonasfj Any suggestions for the cleanest way of using dart files from a third_party folder?

### PR DESCRIPTION
**Pull request by mannprerak2**
Closes #43 
- Added licenses to third_party<folder> with reference to generated files.
- Updated preambles for third-party code

### CODE REVIEW

File: example/c_json/pubspec.yaml [hunk 1]
```diff
@@ -24,3 +24,28 @@ ffigen:
...
+    //
+    // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    // THE SOFTWARE.
+
+    // AUTO GENERATED FILE, DO NOT EDIT.
```
Comment (dcharkes) [inline example/c_json/pubspec.yaml:49]: This should be always added after the preamble I think.

File: third_party/libclang/license.txt [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+court of competent jurisdiction determines that the patent provision (Section
+3), the indemnity provision (Section 9) or other Section of the License
+conflicts with the conditions of the GPLv2, you may retroactively and
+prospectively choose to deem waived or otherwise exclude such Section(s) of
+the License, but only in their entirety and only with respect to the Combined
+Software.
+
+
+Files generated from libclang source code are
+- test/large_integration_tests/_expected_libclang_bindings.dart
```
Comment (dcharkes) [inline third_party/libclang/license.txt:222]: sort alphabetically

Comment (dcharkes) [reply third_party/libclang/license.txt:222]: This is not fixed yet, did you forget to push?

Comment (mannprerak2) [reply third_party/libclang/license.txt]: Oh, my bad, 🙈🙈 pushed it now.

File: example/c_json/cjson_generated_bindings.dart [hunk 3]
```diff
@@ -18,9 +18,9 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// AUTO GENERATED FILE, DO NOT EDIT.
-//
-// Generated by `package:ffigen`.
+/// AUTO GENERATED FILE, DO NOT EDIT.
```
Comment (dcharkes) [inline example/c_json/cjson_generated_bindings.dart:21]: Two slashes should be enough - it's not a DartDoc comment.

File: lib/src/code_generator/writer.dart [hunk 4]
```diff
@@ -105,12 +105,12 @@ class Writer {
...
       s.write(header);
       s.write('\n');
-    } else {
-      // Write default header, in case none was provided.
-      s.write(makeDartDoc(
-          'AUTO GENERATED FILE, DO NOT EDIT.\n\nGenerated by `package:ffigen`.'));
     }
 
+    // Write auto generated declaration.
+    s.write(makeDartDoc(
```
Comment (dcharkes) [inline lib/src/code_generator/writer.dart:111]: Should be `makeDoc(...)`.

**Pull request merged**